docs/cross-contract/callbacks.md

```diff
@@ -4,10 +4,48 @@ sidebar_position: 2
 
 # Callbacks
 
-<!-- What are callbacks -->
+NEAR Protocol is a sharded, proof-of-stake blockchain that behaves differently than proof-of-work blockchains. When interacting with a native Rust (compiled to Wasm) smart contract, cross-contract calls are asynchronous. Callbacks are used to either get the result of a cross-contract call or tell if a cross-contract call has succeeded or failed.
 
-<!-- What do #[callback] decorators do -->
+There are two techniques to write cross-contract calls: [high-level](https://github.com/near/near-sdk-rs/tree/master/examples/cross-contract-high-level) and [low-level](https://github.com/near/near-sdk-rs/tree/master/examples/cross-contract-low-level). This document will mostly focus on the high-level approach. There are two examples in the Rust SDK repository that demonstrate these, as linked above. Note that these examples use cross-contract calls "to itself." We'll show a more common example here, where a simple cross-contract call is made to a whitelist smart contract, returning whether an account is in the whitelist or not.
 
-<!-- Why do we need them and when should they be used -->
+The common pattern with cross-contract calls is to call a method on an external smart contract, use `.then` syntax to specify a callback, and then retrieve the result or status of the promise. The callback will typically live inside the same, calling smart contract. There's a special macro used for the callback function, which is [#[private]](https://docs.rs/near-sdk-core/latest/near_sdk_core/struct.AttrSigInfo.html#structfield.is_private). We'll see this pattern in the example below.
 
-TODO
+The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up that serve as interfaces for the smart contract being called, and (typically) the current contract doing the calling, where callback logic will live. The second trait is, by convention, referred to as `ext_self`. Here are the two traits used in a simple example:
```

Comment:
nit: I was trying to unravel the mid sentence in my mind for a bit so I think its worth rewording it. I believe this is what you meant and hopefully its clearer:

```suggestion
The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up as interfaces for the smart contract being called, and (typically) the current contract doing the calling (where callback logic will live). The second trait is, by convention, referred to as `ext_self`. Here are the two traits used in a simple example:
```

The previous one wasn't clear where the subject of the statement "where callback logic will live" was, IMO.

Reply:
Thank you!